tests/EnumTest.php

```diff
@@ -60,6 +60,14 @@ public function testFailToCreateEnumWithInvalidValueThroughNamedConstructor($val
         EnumFixture::from($value);
     }
 
+    public function testFailToCreateEnumWithEnumItselfThroughNamedConstructor(): void
+    {
+        $this->expectException(\UnexpectedValueException::class);
+        $this->expectExceptionMessage("Value 'foo' is not part of the enum " . EnumFixture::class);
+
+        EnumFixture::from(EnumFixture::FOO());
```

Comment:
I added one more test here to clearly mention from should not work with the enum object itself.